src/core/lib/ConsiderationDecoder.sol

```diff
@@ -921,28 +932,30 @@ contract ConsiderationDecoder {
                     considerationLength := mload(OneWord)
 
                     {
-                        // Calculate total size of offer & consideration arrays.
-                        let totalOfferSize :=
-                            shl(SpentItem_size_shift, offerLength)
-                        let totalConsiderationSize :=
-                            mul(ReceivedItem_size, considerationLength)
-
-                        // Add 4 words to total size to cover the offset and
-                        // length fields of the two arrays.
-                        let totalSize :=
+                        // Calculate end offsets of offer & consideration arrays.
+                        let offerEndOffset :=
                             add(
-                                FourWords,
-                                add(totalOfferSize, totalConsiderationSize)
+                                add(offsetOffer, OneWord),
+                                shl(SpentItem_size_shift, offerLength)
                             )
+                        let considerationEndOffset :=
+                            add(
+                                add(offsetConsideration, OneWord),
+                                mul(ReceivedItem_size, considerationLength)
+                            )
+
                         // Don't continue if returndatasize exceeds 65535 bytes
-                        // or is greater than the calculated size.
+                        // or less than the end offsets.
                         invalidEncoding :=
                             or(
                                 gt(
                                     or(offerLength, considerationLength),
                                     generateOrder_maximum_returndatasize
                                 ),
```

Comment:
This check does not match with the comment. @0age 

The comment says:

```solidity
// Don't continue if returndatasize exceeds 65535 bytes
```

But here we are checking the `or(offerLength, considerationLength)` against `generateOrder_maximum_returndatasize`

Reply:
Do we need to change this?

Reply:
So we definitely still want to keep the constraint on array length just to protect against funky overflow scenarios in the preceding logic; the size of returndata is still somewhat constrained by comparison against the derived offsets as well

I'd say `generateOrder_maximum_returned_array_length` would be more accurate for the name; and comment could just say `// Don't continue if either offer or consideration length exceeds 65535` or something